pydicom/filewriter.py

```diff
@@ -48,75 +140,12 @@ def correct_ambiguous_vr_element(elem, ds, is_little_endian):
-                # If BitsAllocated is > 8 then VR shall be OW,
-                # else may be OB or OW.
-                # If we get here, the data has not been written before,
-                # so we default to OB for BitsAllocated 1 or 8
-                elem.VR = 'OW' if ds.BitsAllocated > 8 else 'OB'
-
-        # 'US or SS' and dependent on PixelRepresentation
-        elif elem.tag in [
-                0x00189810, 0x00221452, 0x00280104, 0x00280105, 0x00280106,
-                0x00280107, 0x00280108, 0x00280109, 0x00280110, 0x00280111,
-                0x00280120, 0x00280121, 0x00281101, 0x00281102, 0x00281103,
-                0x00283002, 0x00409211, 0x00409216, 0x00603004, 0x00603006
-        ]:
-            # US if PixelRepresenation value is 0x0000, else SS
-            #   For references, see the list at
-            #   https://github.com/darcymason/pydicom/pull/298
-            if 'PixelRepresentation' in ds:
-                if ds.PixelRepresentation == 0:
-                    elem.VR = 'US'
-                    byte_type = 'H'
-                else:
-                    elem.VR = 'SS'
-                    byte_type = 'h'
-                elem.value = convert_numbers(elem.value, is_little_endian,
-                                             byte_type)
-
-        # 'OB or OW' and dependent on WaveformBitsAllocated
-        elif elem.tag in [0x54000100, 0x54000112, 0x5400100A, 0x54001010]:
-            # If WaveformBitsAllocated is > 8 then OW, otherwise may be
-            #   OB or OW, however not sure how to handle this.
-            #   See PS3.3 C.10.9.1.
-            if 'WaveformBitsAllocated' in ds:
-                if ds.WaveformBitsAllocated > 8:
-                    elem.VR = 'OW'
-
-        # 'US or OW': 0028,3006 LUTData
-        elif elem.tag in [0x00283006]:
-            if 'LUTDescriptor' in ds:
-                # First value in LUT Descriptor is how many values in
-                #   LUTData, if there's only one value then must be US
-                # As per PS3.3 C.11.1.1.1
-                if ds.LUTDescriptor[0] == 1:
-                    elem.VR = 'US'
-                    elem.value = convert_numbers(elem.value, is_little_endian,
-                                                 'H')
-                else:
-                    elem.VR = 'OW'
-
-        # 'OB or OW': 60xx,3000 OverlayData and dependent on Transfer Syntax
-        elif (elem.tag.group in range(0x6000, 0x601F, 2)
-              and elem.tag.elem == 0x3000):
-            # Implicit VR must be OW, explicit VR may be OB or OW
-            #   as per PS3.5 Section 8.1.2 and Annex A
-            if ds.is_implicit_VR:
-                elem.VR = 'OW'
+        try:
+            _correct_ambiguous_vr_element(elem, ds, is_little_endian)
+        except AttributeError as e:
+            reason = ('Failed to resolve ambiguous VR for tag'
+                      ' {}: '.format(elem.tag)) + str(e)
+            raise AttributeError(reason)
```

Comment:
Won't this still be misleading? Its not the missing tag that has the ambiguous VR.